pkg/api/http/http.go

```diff
@@ -73,6 +73,7 @@ type api struct {
 	sendToOutputBindingFn func(ctx context.Context, name string, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error)
 	readyStatus           bool
 	outboundReadyStatus   bool
+	metricSpec            config.MetricSpec
```

Comment:
Where is this set?

Reply:
fixed

Reply:
Concerned that we didn't have failing tests, even though we weren't setting this field- do we have enough test coverage?

Reply:
I agree! Double-checked tests; this should be better and properly fails when new code is. PTAL

Reply:
Maybe I am missing something but with the current HEAD, the tests still pass for me if I remove setting this field.

Reply:
As we are only using that field this method:

```
// GetHTTPIncreasedCardinality returns true if increased cardinality is enabled for HTTP metrics
func (m MetricSpec) GetHTTPIncreasedCardinality(log logger.Logger) bool {
	if m.HTTP == nil || m.HTTP.IncreasedCardinality == nil {
		// The default is false
		return false
	}
	return *m.HTTP.IncreasedCardinality
}
```

The method returns false ("low cardinality") as m.HTTP is nil. That stops and returns early. If instead of an "or"(||), we had an "And" (&&), we would have a nice panic there. That's why I've missed it :)  (I played with a similar example here https://goplay.tools/snippet/EMgw3OgHLPH) 

If you remove the negation on the comparison here: https://github.com/dapr/dapr/pull/7756/files#diff-a6c70a4296446b8b8cbe6bebb3d298107381d89c750b3bb47177629cd57e6a0bR79 You should get the tests failing.

Reply:
@JoshVanL ping ?

Reply:
Please add a test case which hits this code path, i.e. would panic if this field was omitted. Tests continue to pass for me when this field has been removed.

FYI there looks to be a lint failure https://github.com/dapr/dapr/actions/runs/9508989782/job/26211228529?pr=7756

Reply:
Fixed the linting and moved the field to a pointer --> this should allow us to diff between unset/false.